.github/workflows/build.yaml

```diff
@@ -0,0 +1,53 @@
+name: Build and verify
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches: [$default-branch]
+permissions:
+  contents: read
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        include:
+          - jdk: 11.0.16
+            sonarEnabled: true
+          - jdk: 17.0.4
+            sonarEnabled: false
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3.0.2
+        with:
+          # Fetch entire history for SonarCloud analysis.
+          fetch-depth: 0
+      - name: Set up JDK
+        uses: actions/setup-java@v3.5.1
+        with:
+          java-version: ${{ matrix.jdk }}
+          distribution: temurin
+          cache: maven
+      - name: Cache SonarCloud packages
+        if: ${{ matrix.sonarEnabled }}
+        uses: actions/cache@v3.0.8
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
```

Comment:
Nice. (In a separate PR) we might want to do something similar for Maven dependencies.

(Or are those cached by default? Below we have "Remove installed project artifacts" after all.)

Reply:
Ah, nevermind; I now see the `cache` parameter to `actions/setup-java`.